Mathlib/LinearAlgebra/Matrix/SVD/Defs.lean

```diff
@@ -0,0 +1,448 @@
+`docs#Matrix.isHermitian_mul_conjTranspose_self`. Hence, they are diagonalizable/decomposable as
+`AᴴA = V E Vᴴ` with `E` a matrix of eigenvalues of `AᴴA`. We can apply the reordering/partitioning
+from the last paragraph `eigenColumnEquiv` to the columns of `V`, rows and columns of `E` and rows
+of `Vᴴ`with out changing the value of the product `VEVᴴ`. With this in mind we have:
+    $$AᴴA = V × E × Vᴴ = [V₁ V₂] [μ 0; 0 0] [V₁ᴴ; V₂ᴴ] = V₁μV₁ᴴ$$
+where the matrix `μ` is a diagonal matrix containing the non-zero eigenvalues of the matrix AᴴA
+ordered so as to correspond to the eigenvectors of AᴴA. This is given the name
+`V₁_mul_μ_mul_V₁_conjTranspose` **reduced spectral theorem, right eigenvectors** below.
+
+We can repeat a similar procedure with `AAᴴ = UEUᴴ` and obtain:
+    $$AAᴴ = U × E × Uᴴ = [U₁ U₂] [μ' 0; 0 0] [U₁ᴴ; U₂ᴴ] = U₁μ'U₁ᴴ$$
+where the matrix `μ'` is a diagonal matrix containing the non-zero eigenvalues of the matrix AAᴴ
+ordered so as to correspond to the eigenvectors of AAᴴ. This is given the name
+`U₁'_mul_μ'_mul_U₁'_conjTranspose` **reduced spectral theroem, left eigenvectors**
+
+Note that `μ` and `μ'` contain the same values in different orders.
+
+Note that we chose (arbitrarily) to favor the right-eigenvector order. Hence, the matrix `svdσ` is
+also the diagonal matrix whose entries are the square roots of the diagonals of `svdμ`. That is
+we define `svdξ = [svdσ, 0; 0 0]` with appropriate dimensions for the zero matrices. Further the
+following equation holds: `svdξᵀ * svdξ = [μ, 0; 0 0]`
+
+Lemmas `reduced_spectral_theorem` (`reduced_spectral_theorem'`) shows that AᴴA and AAᴴ can be
+reduced to products containing only the non-zero singular eigenvalue/eigenvectors. This is later
+used in proving the main SVD theorem. A few lemmas are provided about the invertibility of the
+non-zero singular values matrix: `svdσ_inv`, `σ_inv_μ_σ_inv_eq_one`, `IsUnit_det_svdσ`,
+`IsUnit_det_svdσ_mapK` and `svdσ_inv_mapK`.
+
+We also transfer some of the lemmas we need about eigenvector matrices (for example that
+they are unitary: i.e. inverse is conjugate transpose.). Note that since invertibility in mathlib is
+defined for square matrices while our matrices are partitioned i.e. N × (N₁ ⊕ N₂) and N ≃ (N ⊕ N₂)
+Lean cannot apply the Invertible definition. We work around this where necessary.
+
+To make relating left eigenvectors to right eigenvectors easier we define U₁ = AV₁σ⁻¹ while U₂ is
+obtained from the eigenvectors of (AAᴴ). This avoid a lengthy reindexing operation with many proofs.
+The vectors in U₂ have no such issue since they are multiplied by zero singular values anyway.
+
+## Tags
+Singular Value decomposition, SVD
+-/
+
+variable {𝕂 : Type*} [IsROrC 𝕂] [DecidableEq 𝕂]
+variable {M N : ℕ}
+
+open Matrix BigOperators
+
+namespace Matrix
+
+/-- The right eigenvectors of a matrix A corresponding to its non-zero eigenvalues -/
+noncomputable def svdV₁ (A : Matrix (Fin M) (Fin N) 𝕂) : Matrix (Fin N) (Fin (A.rank)) 𝕂 :=
```

Comment:
the names in mathlib have to convey a meaning. In general, they are longer and more expressive than the names we would use in papers (which are just notations), because they have to be compatible with the whole library.